growth-fancy/script.js

```diff
@@ -0,0 +1,243 @@
+let recordLimit = 10000;
+
+let loadingIndicatorElem;
+let loadingZoneDescElem;
+let errorDescElem;
+let flotContainerElem;
+
+function handleError(err) {
+	let displayString = "Error: '" + err + "' - report this to rackover@racknet.noip.me";
+	console.log(err);
+	errorDescElem.innerHTML = displayString;
+	loadingIndicatorElem.style.display = "none";
+}
+
+function setLoadingZoneDesc(str) {
+	loadingZoneDescElem.innerHTML = str;
+}
+
+function appendToLoadingZoneDesc(str) {
```

Comment:
It is no good to modify innerHTML on the fly for such cases. You should prefer `append` or `appendChild` and append it <p> elements for example. The only time you should be accessing innerHTML is to clear the text zone.